### PR TITLE
[core] Use main repo as source of truth

### PR DESCRIPTION
**Pull request by oliviertassinari**
- https://deploy-preview-144--material-ui-x.netlify.app/components/autocomplete/#ComboBox.tsx
- https://deploy-preview-144--material-ui-x.netlify.app/storybook/

### CODE REVIEW

File: docs/pages/_app.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import MyApp from '@material-ui/monorepo/docs/pages/_app';
+
+export default MyApp;
```
Comment (oliviertassinari) [inline docs/pages/_app.js:3]: The beauty of the approach

**Comment by oliviertassinari** [comment]
@dtassone It seems that we could move forward with the approach. There is still a lot to clean up (+123k LOC diff is wrong, should probably be 1/10 that figure), but a possible architecture is in place:

1. Install mui-org/material-ui as a dependency under `@material-ui/monorepo`
2. Use the sources from the mono repo for all the modules we want to share (prettier scripts, eslint config, documentation components, test scripts etc)
3. Have a new and second documentation website, hosted on material-ui-x.netlify.
4. Have the pages we are interested into proxied. We would have `material-ui.com/components/data-grid` proxy to `material-ui-x.netlify.app/components/data-grid`.

Et voilà, we can focus on building features, not infrastructure :D.

**Comment by eps1lon** [comment]
Have you considered using git submodules instead?

**Comment by oliviertassinari** [comment]
> Have you considered using git submodules instead?

@eps1lon I haven't seen git submodules in use since my first internship in the industry in 2014 but that doesn't mean they can't be a better solution to documenting the components. I'm looking into it.

**Comment by eps1lon** [comment]
I used them for https://github.com/eps1lon/material-ui-playroom to use the docs demos. Maybe something similar works too. Probably easier to handle than the indirection with another repo that needs to be published.

**Comment by dtassone** [comment]
I think this is a great temporary solution. 

I tend to think that we should rethink the solution as a whole, Core, docs, and X.

I wouldn't spend more time on this, for now, as the solution will get to where we want.

**Comment by oliviertassinari** [comment]
For the little exploration I did with git submodules, it seems to shine especially in the case where work is required in material-ui in order to accommodate for constraints of material-ui-x. Is this a case we should optimize for? Not sure it will be very frequent. Outside of that, I didn't find anything (yet?) that is significantly better. 

Also, it seems that we can switch between yarn git and git submodules without a massive refactorization, so maybe we could keep the same direction and course correct if we hit something compelling enough to switch?

File: docs/pages/_app.js (hunk 1 above)
Comment (dtassone) [reply docs/pages/_app.js]: How about`export {default as MyApp} from...`? 1 line

**Comment by dtassone** [comment]
Nice one 👍

File: docs/pages/_app.js (hunk 1 above)
Comment (oliviertassinari) [reply docs/pages/_app.js]: As far as I know, named export doesn't work with Next.js custom App: https://nextjs.org/docs/advanced-features/custom-app. 

We could probably do

```jsx
export { default } from '@material-ui/monorepo/docs/pages/_app';
```

But the commit the comment is based on is outdated, the file has grown to:

```jsx
import MyApp from '@material-ui/monorepo/docs/pages/_app';
import { LicenseInfo } from '@material-ui/x-grid';

// Remove the license warning from demonstration purposes
LicenseInfo.setLicenseKey(
  '0f94d8b65161817ca5d7f7af8ac2f042T1JERVI6TVVJLVN0b3J5Ym9vayxFWFBJUlk9MTY1NDg1ODc1MzU1MCxLRVlWRVJTSU9OPTE=',
);

export default MyApp;
```

**Comment by oliviertassinari** [comment]
I'm not sure if the best part of the pull request is the `−2,180` deleted lines (e.g. with [eslintrc](https://github.com/mui-org/material-ui-x/blob/master/.eslintrc.js)) or the fact that we have most of the infrastructure to put the documentation in production.

**Comment by dtassone** [comment]
Infra for doc 👍 
eslintrc cleanup is the extra bit :)

**Comment by oliviertassinari** [comment]
@dtassone Should I migrate the first page, as an example so we can then parallelize the work?

**Comment by dtassone** [comment]
Sounds like a plan 😄

**Pull request merged**